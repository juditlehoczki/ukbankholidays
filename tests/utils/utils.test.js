import {
  formatBankHolidays,
  convertKebabToTitle,
  formatDate,
} from "../../src/utils/utils";
import mockData from "../mockData";

Date.now = jest.fn(() => new Date(2022, 0, 30));

describe("formatBankHolidays", () => {
  test("returns bank holidays in the expected format", () => {
    const actualResult = formatBankHolidays(mockData);
    expect(actualResult).toMatchSnapshot();
  });
  test("returns max 5 bank holidays", () => {
    const actualResult = formatBankHolidays(mockData);
    for (const region in actualResult) {
      expect(actualResult[region].length).toBeLessThanOrEqual(5);
    }
  });
});

describe("convertKebabToTitle", () => {
  test("converts kebab case to title case", () => {
    expect(convertKebabToTitle("england-and-wales")).toBe("England And Wales");
  });
});

describe("formatDate", () => {
  test("formats date string correctly", () => {
    expect(formatDate("2022-01-29")).toBe("29 January 2022");
  });
});
