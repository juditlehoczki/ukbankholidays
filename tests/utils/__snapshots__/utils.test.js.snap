// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatBankHolidays matches snapshot 1`] = `
Object {
  "england-and-wales": Array [
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-04-18",
      "notes": "",
      "title": "Easter Monday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-03",
      "notes": "",
      "title": "Platinum Jubilee bank holiday",
    },
  ],
  "northern-ireland": Array [
    Object {
      "bunting": true,
      "date": "2022-03-17",
      "notes": "",
      "title": "St Patrick’s Day",
    },
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-04-18",
      "notes": "",
      "title": "Easter Monday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
  ],
  "scotland": Array [
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-03",
      "notes": "",
      "title": "Platinum Jubilee bank holiday",
    },
  ],
}
`;

exports[`formatBankHolidays returns bank holidays in the expected format 1`] = `
Object {
  "england-and-wales": Array [
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-04-18",
      "notes": "",
      "title": "Easter Monday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-03",
      "notes": "",
      "title": "Platinum Jubilee bank holiday",
    },
  ],
  "northern-ireland": Array [
    Object {
      "bunting": true,
      "date": "2022-03-17",
      "notes": "",
      "title": "St Patrick’s Day",
    },
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-04-18",
      "notes": "",
      "title": "Easter Monday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
  ],
  "scotland": Array [
    Object {
      "bunting": false,
      "date": "2022-04-15",
      "notes": "",
      "title": "Good Friday",
    },
    Object {
      "bunting": true,
      "date": "2022-05-02",
      "notes": "",
      "title": "Early May bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-02",
      "notes": "",
      "title": "Spring bank holiday",
    },
    Object {
      "bunting": true,
      "date": "2022-06-03",
      "notes": "",
      "title": "Platinum Jubilee bank holiday",
    },
  ],
}
`;
